datastore/datastore_record.go

```diff
@@ -12,22 +12,20 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-// Package record describes Metric Records that store some data about each imported metric in App Engine Datastore.
-package record
+package datastore
 
 import (
 	"context"
 	"fmt"
-	"time"
-
 	"google.golang.org/appengine/datastore"
 	"google.golang.org/appengine/log"
+	"time"
 )
 
-//go:generate mockgen -destination=../mocks/mock_metric_record.go -package=mocks github.com/google/ts-bridge/record MetricRecord
+//go:generate mockgen -destination=../mocks/mock_metric_record.go -package=mocks github.com/google/ts-bridge/datastore MetricRecord
 
-// MetricRecord is an interface implemented by DatastoreMetricRecord.
-type MetricRecord interface {
+// MetricRecord is an interface implemented by StoredMetricRecord.
+type MetricRecord = interface {
```

Comment:
Remove the `=`

Reply:
Also I think the MetricRecord interface should be defined outside the datastore specific code, perhaps `record/record.go`, because it will be implemented by the Datastore code as well as future backends.

Reply:
@dparrish So this use of Go 1.9 alias is intentional, since otherwise I get:

```
# github.com/google/ts-bridge/app
app/main.go:166:18: cannot use "github.com/google/ts-bridge/datastore".New() (type *"github.com/google/ts-bridge/datastore".Manager) as type storage.Manager in assignment:
	*"github.com/google/ts-bridge/datastore".Manager does not implement storage.Manager (wrong type for NewMetricRecord method)
		have NewMetricRecord("context".Context, string, string) ("github.com/google/ts-bridge/datastore".MetricRecord, error)
		want NewMetricRecord("context".Context, string, string) (storage.MetricRecord, error)
app/main.go:169:18: cannot use "github.com/google/ts-bridge/datastore".New() (type *"github.com/google/ts-bridge/datastore".Manager) as type storage.Manager in assignment:
	*"github.com/google/ts-bridge/datastore".Manager does not implement storage.Manager (wrong type for NewMetricRecord method)
		have NewMetricRecord("context".Context, string, string) ("github.com/google/ts-bridge/datastore".MetricRecord, error)
		want NewMetricRecord("context".Context, string, string) (storage.MetricRecord, error)
FAIL	github.com/google/ts-bridge/tsbridge [build failed]

Process finished with exit code 2

# github.com/google/ts-bridge/tsbridge [github.com/google/ts-bridge/tsbridge.test]
tsbridge/config_test.go:34:81: cannot use "github.com/google/ts-bridge/datastore".New() (type *"github.com/google/ts-bridge/datastore".Manager) as type storage.Manager in field value:
	*"github.com/google/ts-bridge/datastore".Manager does not implement storage.Manager (wrong type for NewMetricRecord method)
		have NewMetricRecord("context".Context, string, string) ("github.com/google/ts-bridge/datastore".MetricRecord, error)
		want NewMetricRecord("context".Context, string, string) (storage.MetricRecord, error)
tsbridge/config_test.go:73:97: cannot use "github.com/google/ts-bridge/datastore".New() (type *"github.com/google/ts-bridge/datastore".Manager) as type storage.Manager in field value:
	*"github.com/google/ts-bridge/datastore".Manager does not implement storage.Manager (wrong type for NewMetricRecord method)
		have NewMetricRecord("context".Context, string, string) ("github.com/google/ts-bridge/datastore".MetricRecord, error)
		want NewMetricRecord("context".Context, string, string) (storage.MetricRecord, error)
tsbridge/metric_test.go:123:86: cannot use "github.com/google/ts-bridge/datastore".New() (type *"github.com/google/ts-bridge/datastore".Manager) as type storage.Manager in argument to NewMetric:
	*"github.com/google/ts-bridge/datastore".Manager does not implement storage.Manager (wrong type for NewMetricRecord method)
		have NewMetricRecord("context".Context, string, string) ("github.com/google/ts-bridge/datastore".MetricRecord, error)
		want NewMetricRecord("context".Context, string, string) (storage.MetricRecord, error)
tsbridge/metric_test.go:163:84: cannot use "github.com/google/ts-bridge/datastore".New() (type *"github.com/google/ts-bridge/datastore".Manager) as type storage.Manager in argument to NewMetric:
	*"github.com/google/ts-bridge/datastore".Manager does not implement storage.Manager (wrong type for NewMetricRecord method)
		have NewMetricRecord("context".Context, string, string) ("github.com/google/ts-bridge/datastore".MetricRecord, error)
		want NewMetricRecord("context".Context, string, string) (storage.MetricRecord, error)
```

Perhaps I'm missing something here though?

Reply:
Yep, you're hiding incomplete code. datastore.Manager (the struct) does not have all the methods necessary to implement the storage.Manager interface, so the compiler says you're doing it wrong.

You need to add the missing methods to datastore.Manager

Reply:
Ah, I see - fixed up the `StoredMetricRecord` and removed the interface. Thanks!